spec/rspec/expectations/block_snippet_extractor_spec.rb

```diff
@@ -0,0 +1,267 @@
+        let(:expression) do
+          target_method { 1.positive? }; another_method { 2.negative? }
+        end
+
+        it 'correctly extracts the target snippet' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context "when there's another method invocation with the same name on the same line before the target" do
+        let(:expression) do
+          dummy_object.target_method { 2.negative? }; target_method { 1.positive? }
+        end
+
+        let(:dummy_object) do
+          double('dummy_object', :target_method => nil)
+        end
+
+        it 'raises AmbiguousTargetError' do
+          expect { body_content_lines }.to raise_error(BlockSnippetExtractor::AmbiguousTargetError)
+        end
+      end
+
+      context "when there's another method invocation with the same name on the same line after the target" do
+        let(:expression) do
+          target_method { 1.positive? }; dummy_object.target_method { 2.negative? }
+        end
+
+        let(:dummy_object) do
+          double('dummy_object', :target_method => nil)
+        end
+
+        it 'raises AmbiguousTargetError' do
+          expect { body_content_lines }.to raise_error(BlockSnippetExtractor::AmbiguousTargetError)
+        end
+      end
+
+      context "when there's another method invocation with the same name without block on the same line before the target" do
+        let(:expression) do
+          dummy_object.target_method; target_method { 1.positive? }
+        end
+
+        let(:dummy_object) do
+          double('dummy_object', :target_method => nil)
+        end
+
+        it 'correctly extracts the target snippet' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context "when there's another method invocation with the same name without block on the same line after the target" do
+        let(:expression) do
+          target_method { 1.positive? }; dummy_object.target_method
+        end
+
+        let(:dummy_object) do
+          double('dummy_object', :target_method => nil)
+        end
+
+        it 'correctly extracts the target snippet' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'when a hash is given as an argument' do
+        let(:expression) do
+          target_method({ :key => "value" }) { 1.positive? }
+        end
+
+        it 'correctly extracts the target snippet' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'when another method invocation with block is given as an argument' do
+        let(:expression) do
+          target_method(another_method { 2.negative? }) { 1.positive? }
+        end
+
+        it 'correctly extracts the target snippet' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context "when wrong proc is given" do
```

Comment:
I meant something "malicious", but the example is actually testing same behavior with the below example `context "when the block literal is described on different line with the method invocation"`. So I'll remove this example.